doc/rst/technotes/initequals.rst

```diff
@@ -0,0 +1,268 @@
+-------------------------
+
+The ``init=`` method is in many ways similar to the ``init`` method, and they
+share the same semantic rules within the body of the method. For example,
+fields must be initialized in declaration order, and must be initialized unless
+they have a default value.
+
+Where ``init=`` differs is in its invocation. The ``init=`` method will only be
+invoked in two cases: copy initialization and initializing a variable from an
+expression. The ``init=`` method will only be invoked with a single value.
+
+The ``init=`` method may also invoke other initializers through
+``this.init(...)``, but currently may not invoke other ``init=`` methods.
+
+Classes do not support the ``init=`` method because classes will not be
+copy-initialized by the compiler.
+
+The 'init=' Method for Non-Generic Types
+-----------------------------------------
+
+The compiler-generated ``init=`` method for non-generic types is simple. It
+accepts one argument of the same type:
+
+.. code-block:: chapel
+
+  record R {
+    var x : int;
+  }
+
+  // identical to compiler-generated implementation
+  proc R.init=(other : R) {
+    this.x = other.x;
+  }
+
+In order to override this compiler-generated implementation, the user must
+implement an ``init=`` method that can accept an argument of the same type.
+Other user-defined ``init=`` methods will not prevent the compiler from
+generating a default implementation for ``init=``. For example:
+
+.. code-block:: chapel
+
+  record R {
+    var x : int;
+  }
+
+  proc R.init=(other : int) {
+    this.x = other;
+  }
+
+  var A = new R(10);   // compiler-generated initializer 
+  var B = A;           // B.init=(A) , the compiler-generated ``init=``
+  var C : R = 10;      // C.init=(10) , user-defined ``init=``
+  // var D : R = "hello"; // D.init=("hello") , unresolved call!
+
+
+The 'init=' Method for Generic Types
+------------------------------------
+
+The compiler-generated 'init='
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The compiler-generated ``init=`` method for generic types requires knowing the
+intended instantiation in order to disallow copy-initialization from different
+types. In the following example, there should be a compile-time error when
+attempting to initialize a ``R(int)`` from a ``R(real)``.
+
+.. code-block:: chapel
+
+  record R {
+    type T;
+    var x : T;
+  }
+
+  var x : R(real);
+  var y : R(int) = x;
+
+This is accomplished by allowing ``init=`` to query the intended instantiation
+through the expression ``this.type``. The compiler-generated ``init=`` for
+type ``R`` looks like:
+
+.. code-block:: chapel
+
+  proc R.init=(other : this.type) {
+    this.T = other.T;
+    this.x = other.x;
+  }
+
+The first line of this ``init=`` may seem unnecessary, since ``this.type`` must
+already be known. The line ``this.T = other.T`` currently serves as an
+assertion, and future releases may remove the need for such field
+initializations within a generic ``init=``.
```

Comment:
Maybe instead of "currently serves as an assertion" you could write "is currently used by the compiler to ensure that the types match" or something along those lines? I'm hoping it'll say more about what is being asserted.